livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -803,10 +805,18 @@ class ChannelControllerImpl(
         _members.postValue(copy)
     }
 
+    fun upsertMembers(members: List<Member>) {
+        val channelMembers = _members.value ?: mutableMapOf()
+        members.forEach {
+            channelMembers[it.user.id] = it
+        }
+        _members.postValue(channelMembers)
+    }
+
     fun upsertMember(member: Member) {
-        val copy = _members.value ?: mutableMapOf()
-        copy[member.user.id] = member
-        _members.postValue(copy)
+        val channelMembers = _members.value ?: mutableMapOf()
+        channelMembers[member.user.id] = member
+        _members.postValue(channelMembers)
```

Comment:
What do you thing to reuse the previous function?
Then we have the "matching logic" only one place.
You could dome something like:

```suggestion
        upsertMembers(listOf(member))
```